### PR TITLE
More Timeline Semaphore validation

### PR DESCRIPTION
**Pull request by jasuarez**
This PR add more timeline semaphores VUIDs validation, as well as new tests for some of them.

Specifically, it tries to track correctly the status of the semaphores.

### CODE REVIEW

**Comment by mark-lunarg** [comment]
@jasuarez, looks like most of the files need a clang-format pass (please do the commits individually -- if you can't run this tool let me know and I'll take care of it this time).  Thanks!

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -2275,16 +2281,14 @@ bool CoreChecks::PreCallValidateQueueSubmit(VkQueue queue, uint32_t submitCount,
     for (uint32_t submit_idx = 0; submit_idx < submitCount; submit_idx++) {
         const VkSubmitInfo *submit = &pSubmits[submit_idx];
         auto *info = lvl_find_in_chain<VkTimelineSemaphoreSubmitInfoKHR>(submit->pNext);
-        for (uint32_t i = 0; i < submit->waitSemaphoreCount; ++i) {
+        for (uint32_t i = 0; info && i < submit->waitSemaphoreCount; ++i) {
```
Comment (mark-lunarg) [inline layers/core_validation.cpp]: Would it hurt (not sure!) to put these loops inside an `if (info) {}` section (here and below on 9450)?  This gives me the shivers.  Happy to do this if it's OK...

Comment (jasuarez) [reply layers/core_validation.cpp]: Not problem at all! Pushing an updated version.

**Comment by mark-lunarg** [comment]
@jasuarez, thanks so much!  After we get a green board it'll go.  And we all appreciate you jumping back on these issues

**Pull request merged**